File: tests/algorithms/x/Java/greedy_methods/minimum_waiting_time.java
public class Main {

    static int[] insertion_sort(int[] a) {
        int i = 1;
        while (i < a.length) {
            int key = a[i];
            int j = i - 1;
            while (j >= 0 && a[j] > key) {
a[j + 1] = a[j];
                j = j - 1;
            }
a[j + 1] = key;
            i = i + 1;
        }
        return a;
    }

    static int minimum_waiting_time(int[] queries) {
        int n = queries.length;
        if (n == 0 || n == 1) {
            return 0;
        }
        int[] sorted = ((int[])(insertion_sort(((int[])(queries)))));
        int total = 0;
        int i_1 = 0;
        while (i_1 < n) {
            total = total + sorted[i_1] * (n - i_1 - 1);
            i_1 = i_1 + 1;
        }
        return total;
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            System.out.println(minimum_waiting_time(((int[])(new int[]{3, 2, 1, 2, 6}))));
            System.out.println(minimum_waiting_time(((int[])(new int[]{3, 2, 1}))));
            System.out.println(minimum_waiting_time(((int[])(new int[]{1, 2, 3, 4}))));
            System.out.println(minimum_waiting_time(((int[])(new int[]{5, 5, 5, 5}))));
            System.out.println(minimum_waiting_time(((int[])(new int[]{}))));
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }
}

File: tests/algorithms/x/PHP/greedy_methods/minimum_waiting_time.php
<?php
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
$__start_mem = memory_get_usage();
$__start = _now();
  function insertion_sort($a) {
  $i = 1;
  while ($i < count($a)) {
  $key = $a[$i];
  $j = $i - 1;
  while ($j >= 0 && $a[$j] > $key) {
  $a[$j + 1] = $a[$j];
  $j = $j - 1;
};
  $a[$j + 1] = $key;
  $i = $i + 1;
};
  return $a;
};
  function minimum_waiting_time($queries) {
  $n = count($queries);
  if ($n == 0 || $n == 1) {
  return 0;
}
  $sorted = insertion_sort($queries);
  $total = 0;
  $i = 0;
  while ($i < $n) {
  $total = $total + $sorted[$i] * ($n - $i - 1);
  $i = $i + 1;
};
  return $total;
};
  echo rtrim(json_encode(minimum_waiting_time([3, 2, 1, 2, 6]), 1344)), PHP_EOL;
  echo rtrim(json_encode(minimum_waiting_time([3, 2, 1]), 1344)), PHP_EOL;
  echo rtrim(json_encode(minimum_waiting_time([1, 2, 3, 4]), 1344)), PHP_EOL;
  echo rtrim(json_encode(minimum_waiting_time([5, 5, 5, 5]), 1344)), PHP_EOL;
  echo rtrim(json_encode(minimum_waiting_time([]), 1344)), PHP_EOL;
$__end = _now();
$__end_mem = memory_get_peak_usage();
$__duration = max(1, intdiv($__end - $__start, 1000));
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;
